Kernel/SubstHelper.hpp

```diff
@@ -26,6 +26,45 @@ namespace Kernel {
 
 using namespace Lib;
 
+struct SubstApplicator {
+  virtual ~SubstApplicator() = default;
+  virtual TermList operator()(unsigned v) const = 0;
+};
+
+/**
+ * Term with a substitution applied to it lazily.
+ */
+struct AppliedTerm
+{
+  TermList term;
+  bool aboveVar;
```

Comment:
Could you maybe document the intuition of `aboveVar`? Maybe an example would be helpful.

Reply:
I'll add an example.